dsl/src/test/scala/com/crobox/clickhouse/dsl/JoinQueryIT.scala

```diff
@@ -82,73 +105,63 @@ class JoinQueryIT
   ) { (joinType, result) =>
     it should s"join correctly on: $joinType" in {
       assumeMinimalClickhouseVersion(20)
-      val query: OperationalQuery =
-        select(itemId).from(
-          select(itemId)
-            .from(TwoTestTable)
-            .asOfJoin(joinType, ThreeTestTable, Option("TTT"), (col2, "<="))
-            .using(itemId)
-        )
-      val resultRows = chExecutor.execute[Result](query).futureValue.rows
+
+      var query: OperationalQuery =
+        select(itemId, col2)
+          .from(TwoTestTable)
+          .where(notEmpty(itemId))
+          .join(joinType, ThreeTestTable)
+          .on((itemId, "=", itemId), (col2, "<=", col2))
```

Comment:
maybe a DSL like `itemId.joinOn(itemId), col2.joinOnSmaller(col2)` would more clear? Also it prevents the `String` items that you want to prevent to keep it SQL safe.

I'm not completly sure what the naming should be though

Reply:
Since 'AS OF' joins won't be used that often (very, very custom clickhouse) the default behavior is to join on simple columns, which is supported